### DISCUSSION TITLE
What is Flux v2 and how is it accessed?

### DISCUSSION THREAD

**Discussion by squaremo** [fluxcd/flux2#178] [General]
It [is said](https://toolkit.fluxcd.io/):

> Flux v2 [...] will be assembled from the GitOps Toolkit components.

What is the nature of the assembled Flux v2, and how do you actually use it? (Secondarily: what becomes of the fluxcd/flux repository?)

Here are some possible answers:

 * Flux v2 is just the GitOps Toolkit, there's no extra thing
 * Flux v2 is an "on rails" mode for the GitOps Toolkit
    - .. accessed with its own CLI
    - .. accessed with a subcommand of `tk` (aka `gotk`)
    - .. available as a written guide
 * Flux v2 builds on GOTK with its own abstractions and command-line tool

**Comment by monadic** [comment]
+1 for "on rails"

**Reply by hiddeco** [reply]
-1 for “accessed with its own CLI” option

**Comment by squaremo** [comment]
Whatever the technical answer, I think the name of this project should be prominent. It would be counter productive to abandon or bury "Flux" when it's the name people know -- or, indeed, to have competing names occupying the same role, which would be a recipe for confusion.

So for example, if the answer to the first part of the question ("What is Flux v2") is:

> Flux v2 is an "on rails" mode for the GitOps Toolkit

then the answer to the second part should be "you access it with a command-line tool named `flux`" (or some variation), and not "you access it with a subcommand of `tk`".

**Reply by hiddeco** [reply]
Taking the Linkerd project as an example, I think we should abandon the `fluxcd/flux` repository in any case (and _maybe_ rename `fluxcd/toolkit` to `fluxcd/flux2`, and `gotk` to `flux`).

My thoughts are based on the following:

- It will make it possible to archive the repository once we do no longer offer support for the project, nor plan to make any more releases. This will result in people no longer being able to open up issues or PRs on the project, freeing us from maintainer burdens.
- I have an (maybe false) assumption that without moving houses, people will continue to post issues for Flux v1 after the support period. Moving creates (hopefully) an extra barrier to prevent this.
- If we decide to extend the support period due to _[insert reason]_, we have a clear separation of issues and PRs.

**Reply by squaremo** [reply]
Needing to keep the issues and PRs and other accoutrements separate is a really good point. It would be chaotic having issues of the new thing intermixed with flux v1 issues, can you imagine? We struggle keeping things organised as it is.

Loth as I am to consign github stars to the dustbin, I'm inclined to agree with you that the fluxcd/flux repo should not be reassigned, and should be retired when flux v1 is.

**Comment by stefanprodan** [comment]
The naming proposal should answer the question "What is Flux v2" https://github.com/fluxcd/toolkit/discussions/367
